src/textual/validation.py

```diff
@@ -0,0 +1,508 @@
+            value: The value to validate.
+
+        Returns:
+            The result of the validation.
+        """
+        # First, check that we're dealing with a number in the range.
+        number_validation_result = super().validate(value)
+        if not number_validation_result.is_valid:
+            return number_validation_result
+
+        # We know it's a number, but is that number an integer?
+        is_integer = float(value).is_integer()
+        if not is_integer:
+            return ValidationResult.failure([Integer.NotAnInteger(self, value)])
+
+        return self.success()
+
+    def describe_failure(self, failure: Failure) -> str | None:
+        """Describes why the validator failed.
+
+        Args:
+            failure: Information about why the validation failed.
+
+        Returns:
+            A string description of the failure.
+        """
+        if isinstance(failure, Integer.NotAnInteger):
+            return f"Must be a valid integer."
+        elif isinstance(failure, Integer.NotInRange):
+            minimum = self.minimum if self.minimum is not None else "-∞"
+            maximum = self.maximum if self.maximum is not None else "∞"
+            return f"Must be in range [{minimum}, {maximum}]."
+        else:
+            return None
+
+
+class Length(Validator):
+    """Validate that a string is within a range (inclusive)."""
+
+    def __init__(
+        self,
+        minimum: int | None = None,
+        maximum: int | None = None,
+        failure_description: str | None = None,
+    ) -> None:
+        super().__init__(failure_description=failure_description)
+        self.minimum = minimum
+        """The inclusive minimum length of the value, or None if unbounded."""
+        self.maximum = maximum
+        """The inclusive maximum length of the value, or None if unbounded."""
+
+    class TooShort(Failure):
+        """Indicates a failure due to the value being too short."""
+
+    class TooLong(Failure):
+        """Indicates a failure due to a value being too long."""
+
+    def validate(self, value: str) -> ValidationResult:
+        """Ensure that value falls within the maximum and minimum length constraints.
+
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The result of the validation.
+        """
+        if self.minimum is not None and len(value) < self.minimum:
+            return ValidationResult.failure([Length.TooShort(self, value)])
+
+        if self.maximum is not None and len(value) > self.maximum:
+            return ValidationResult.failure([Length.TooLong(self, value)])
+
+        return self.success()
+
+    def describe_failure(self, failure: Failure) -> str | None:
+        """Describes why the validator failed.
+
+        Args:
+            failure: Information about why the validation failed.
+
+        Returns:
+            A string description of the failure.
+        """
+        if isinstance(failure, Length.TooShort):
```

Comment:
The error message only reveals part of the requirements. Could we have a single error message that includes the minimum *and* maximum? Should be obvious in context wether the user was too high or too low.